angular/src/directives/CipherListVirtualScroll.directive.ts

```diff
@@ -0,0 +1,101 @@
+    private checkItemSizeCallback: any;
+    private timeout: any;
+
+    constructor(itemSize: number, minBufferPx: number, maxBufferPx: number, checkItemSizeCallback: any) {
+        super(itemSize, minBufferPx, maxBufferPx);
+        this.checkItemSizeCallback = checkItemSizeCallback;
+    }
+
+    onContentRendered() {
+        if (this.timeout != null) {
+            clearTimeout(this.timeout);
+        }
+
+        this.timeout = setTimeout(this.checkItemSizeCallback, 500);
+    }
+}
+
+export function _cipherListVirtualScrollStrategyFactory(cipherListDir: CipherListVirtualScroll) {
+    return cipherListDir._scrollStrategy;
+}
+
+@Directive({
+    selector: 'cdk-virtual-scroll-viewport[defaultItemSize]',
+    providers: [{
+        provide: VIRTUAL_SCROLL_STRATEGY,
+        useFactory: _cipherListVirtualScrollStrategyFactory,
+        deps: [forwardRef(() => CipherListVirtualScroll)],
+    }],
+})
+export class CipherListVirtualScroll implements OnChanges {
+    // tslint:disable
+    static ngAcceptInputType_defaultItemSize: NumberInput;
+    static ngAcceptInputType_minBufferPx: NumberInput;
+    static ngAcceptInputType_maxBufferPx: NumberInput;
+    // tslint:enable
+
+    /** The default size of the items in the list (in pixels). */
+    @Input()
+    get defaultItemSize(): number { return this._defaultItemSize; }
+    set defaultItemSize(value: number) { this._defaultItemSize = coerceNumberProperty(value); }
+    _defaultItemSize = 20;
+
+    /**
+     * The minimum amount of buffer rendered beyond the viewport (in pixels).
+     * If the amount of buffer dips below this number, more items will be rendered. Defaults to 100px.
+     */
+    @Input()
+    get minBufferPx(): number { return this._minBufferPx; }
+    set minBufferPx(value: number) { this._minBufferPx = coerceNumberProperty(value); }
+    _minBufferPx = 100;
+
+    /**
+     * The number of pixels worth of buffer to render for when rendering new items. Defaults to 200px.
+     */
+    @Input()
+    get maxBufferPx(): number { return this._maxBufferPx; }
+    set maxBufferPx(value: number) { this._maxBufferPx = coerceNumberProperty(value); }
+    _maxBufferPx = 200;
+
+    /**
+     * The current size of items in the list (in pixels). This property is updated using checkAndUpdateItemSize.
+     */
+    itemSize: number = this.defaultItemSize;
+
+    checkAndUpdateItemSize = () => {
+        const sampleItem = document.querySelector('cdk-virtual-scroll-viewport .virtual-scroll-item') as HTMLElement;
+        const newItemSize = sampleItem?.offsetHeight;
+
+        if (newItemSize != null && newItemSize !== this.itemSize) {
+            this.itemSize = newItemSize;
+            this._scrollStrategy.updateItemAndBufferSize(this.itemSize, this.minBufferPx, this.maxBufferPx);
+        }
+    }
+
+    ngOnChanges() {
+        this._scrollStrategy.updateItemAndBufferSize(this.itemSize, this.minBufferPx, this.maxBufferPx);
```

Comment:
Is this being re-calculated any time the list changes? wouldn't it be sufficient to just do it on load?